### ISSUE TITLE
feature request: add "Select" mode in Key bindings

### ISSUE THREAD

**Issue by marcozaccari** [alacritty/alacritty#5036]
The key binding configuration is a very powerful tool with a depth level of customization, 
but it fails to discriminate key bindings when the user is selecting text (NormalSelection, LineSelection, BlockSelection).

### Proposal

- add `Select` in `mode` list: `[AppCursor, AppKeypad, Search, Alt, Vi, Select]`

### Benefits

1- Specific key bindings for vi+selection mode only;
2- Avoid unwanted key bindings in vi+selection mode.

#### Example 1

In Vi mode, when selecting text, press Return to copy and exit from visual mode:

```
key_bindings:
  - { key: Return,  mode: Vi|Select, action: Copy          }
  - { key: Return,  mode: Vi|Select, action: ToggleViMode  }
```

#### Example 2

Select text in Vi mode using Shift + Left/Right/Up/Down (like non-vi text editors):

```
key_bindings:
  # Shift+arrows move the cursor in vi mode
  - { key: Left,   mods: Shift,  mode: Vi, action: Left  }
  - { key: Right,  mods: Shift,  mode: Vi, action: Right }
  - { key: Up,     mods: Shift,  mode: Vi, action: Up    }
  - { key: Down,   mods: Shift,  mode: Vi, action: Down  }

  # Shift+arrows start selection if it's not started yet
  - { key: Left,   mods: Shift,  mode: Vi|~Select, action: ToggleNormalSelection  }
  - { key: Right,  mods: Shift,  mode: Vi|~Select, action: ToggleNormalSelection  }
  - { key: Up,     mods: Shift,  mode: Vi|~Select, action: ToggleNormalSelection  }
  - { key: Down,   mods: Shift,  mode: Vi|~Select, action: ToggleNormalSelection  }
```

**Comment by chrisduerr** [comment]
These usecases aren't really selling me on the idea. They mostly seem to be coming from the perspective of not actually wanting to use a "vi" mode, so I'm not sure if trying to shoehorn the vi mode into something different makes much sense here.

Unless there's an actual good uscase, I don't really think we should be adding more things to this just for the sake of it.

**Comment by marcozaccari** [comment]
Sorry if my request looks like a workaround, it was not my intention.
Vi mode is ok, but sometimes we have slightly different workflows.
In fact, the first example refers to **screen**, which offers the convenience of using the same key to start/end text selection:

Ctrl + [, A (enter "copy mode")
move cursor, Return (start selection)
move cursor, Return (end selection + copy to clipboard + exit copy mode)

In this case it is essential to be able to discriminate when we are in selection mode.
Other similar workflows:

**byobu**:
F7 (vi mode)
move cursor, Space (starting selection)
move cursor, Return (end selection, copy to clipboard, exit vi mode)

**tmux**:
Ctrl + [, A (enter vi mode)
move cursor, Ctrl + Space (starting selection)
move cursor, Ctrl + W (end selection, copy to clipboard, exit vi mode)

In cases like these it can happen that we get a collision with a shortcuts used in normal vi mode (Return is still used as `Open` action).

The second example in my previous post, on the other hand, helps those who frequently use CUA editors (vscode, intellij*, ...) and prefer to keep the usual shorcuts shift+arrows in Vi mode.

Thanks for your patience. :)

**Comment by chrisduerr** [comment]
I don't think this is going to be of much use to be honest. On the other hand I believe the patch should be quite simple. So it might be fine to just add it, though I'm not interested in doing it myself.
